bot/cogs/search.py

```diff
@@ -149,120 +151,187 @@ async def anime(self, ctx: Context, *, query: str) -> None:
+                    ```
+                    """
+                )
+                await ctx.send(template)
+
+    @command()
+    async def weather(self, ctx: Context, *, city: str = None) -> None:
+        """
+        Sends current weather in the given city name.
+
+        eg. `weather london`
+        """
+        try:
+            url_formatted_city = city.replace(" ", "-")
+        except CommandInvokeError:
+            ctx.send("You didn't provide a city")
+
+        async with aiohttp.ClientSession() as session:
+            weather_lookup_url = f"https://api.openweathermap.org/data/2.5/weather?q={url_formatted_city}&appid={os.getenv('WEATHER_API_KEY')}"
+            async with session.get(weather_lookup_url) as resp:
+                data = await resp.json()
+
+        if data["cod"] == "401":
+            await ctx.send("Invalid API key")
+            return
+        if data["cod"] == "404":
+            await ctx.send("Invalid city name")
+            return
+
+        weather_embed = Embed(
+            title=f"Current Weather in {city.capitalize()}",
+            color=Color.blue()
+        )
+        longtitude = data["coord"]["lon"]
+        lattitude = data["coord"]["lat"]
+        weather_embed.add_field(
+            name="❯❯Coordinates",
+            value=f"**Longtitude: **`{longtitude}`\n**Latittude: **`{lattitude}`"
+        )
+        actual_temp = round(data["main"]["temp"] / 10, 1)
+        feels_like = round(data["main"]["feels_like"] / 10, 1)
+        weather_embed.add_field(
+            name="❯❯Temperature",
+            value=f"**Temperature: **`{actual_temp}°C`\n**Feels Like: **`{feels_like}°C`"
+        )
+        wind_speed = data["wind"]["speed"]
+        wind_direction = data["wind"]["deg"]
+        weather_embed.add_field(
+            name="❯❯Wind",
+            value=f"**Speed: **`{wind_speed}km/h`\n**Direction: **`{wind_direction}°`",
+        )
+        visibility = round(data["visibility"] / 1000, 2)
+        humidity = data["main"]["humidity"]
+        weather_description = data["weather"][0]["description"]
+        weather_embed.add_field(
+            name="❯❯Miscellaneous",
+            value=f"**Humidity: **`{humidity}%`\n**Visibility: **`{visibility}km`\n**Weather Summary: **`{weather_description}`",
+        )
+
+        if "wind" in weather_description:
+            weather_embed.set_image(url=config.WEATHER_ICONS["wind"])
+        elif "partly" in weather_description:
+            weather_embed.set_image(url=config.WEATHER_ICONS["partly"])
+        elif "cloud" in weather_description:
+            weather_embed.set_image(url=config.WEATHER_ICONS["cloud"])
+        elif "snow" in weather_description:
+            weather_embed.set_image(url=config.WEATHER_ICONS["snow"])
+        elif "rain" in weather_description:
+            weather_embed.set_image(url=config.WEATHER_ICONS["rain"])
+        else:
+            weather_embed.set_image(url=config.WEATHER_ICONS["sun"])
```

Comment:
We should always try to keep the code "DRY" it's a common rule in coding and it refers to: Don't Repeat Yourself.
We don't need to do the same thing over and over again if there's a shorter and usually simpler way.

This specifically can be solved with making a list of possible states iterating over them and if they pass the given condition we'll run the needed function and break the for loop, breaking is essential for this to work since we need to have a fallback to a default state if none of the states matches, this is really easy in python without a need to use extra variable (as you would in most other languages) since you can put an `else` statement to the `for` loop, the code in this else statement will only get called if the `for` loop wasn't broken at any time
```suggestion
        states = ["wind", "partly", "cloud", "snow", "rain"]
        for state in states:
            if state in weather_description:
                weather_embed.set_image(url=config.WEATHER_ICONS[state])
                break
        else:
            weather_embed.set_image(url=config.WEATHER_ICONS["sun"])
 ```